packages/fcl-wc/src/service.js

```diff
@@ -120,17 +119,26 @@ async function connectWc(onClose, {service, client, pairing}) {
       requiredNamespaces,
     })
 
-    if (uri) {
-      if (isMobile()) {
-        const queryString = new URLSearchParams({uri: uri}).toString()
-        // need deep link url for mobile
-        let url = "" + queryString
-        window.open(url, "blank").focus()
-      } else {
-        QRCodeModal.open(uri, () => {
-          onClose()
-        })
-      }
+    const appLink = service.uid || pairing?.peerMetadata?.url
+
+    if (!isMobile() && !pairing) {
+      QRCodeModal.open(uri, () => {
+        onClose()
+      })
+    } else if (!isMobile() && pairing) {
+      log({
+        title: "WalletConnect Session request",
+        message: `
+          ${pairing.peerMetadata.name}
+          Pairing exists, Approve Session in your Mobile Wallet
+        `,
+        level: 2,
+      })
+      sessionRequestHook && sessionRequestHook(pairing.peerMetadata)
```

Comment:
Whats the purpose of this line?

Reply:
If the DApp includes a `sessionRequestHook`, it will be called here to allow handling on the UI side. Since no QR code is shown when a wallet has already been paired, the request just shows up in the mobile wallet.

![Screen Shot 2022-09-01 at 9 54 07 AM](https://user-images.githubusercontent.com/366692/187969966-af4334e3-ac73-46be-974b-d2a1cbbee26b.png)